pkg/frontend/transport/handler.go

```diff
@@ -448,6 +454,12 @@ func writeServiceTimingHeader(queryResponseTime time.Duration, headers http.Head
 	}
 }
 
+func writeStatsHeader(headers http.Header, stats *querier_stats.Stats) {
+	if stats != nil {
+		headers.Set(MimirQueryStatsHeaderName, fmt.Sprintf("fetched_chunk_bytes=%d", stats.GetFetchedChunkBytes()))
```

Comment:
The stats are modified using atomics - so they should be read with: https://github.com/grafana/mimir/blob/e238310c6063b642e4d4007fbba01f6ef1ff532d/pkg/querier/stats/stats.go#L86